### ISSUE TITLE
The response body that is returned from the server after creating a user must contain the authentication token in the header #9

### ISSUE THREAD

**Issue by rasikraj01** [rasikraj01/vogue#1]
When a signup request is sent to the server via the react form it must return an authentication token in the header. When I use postman to check the same I receive the header but if I console log the response body when the request is sent via the browser then the response header does not contain the authentication token.

**Comment by marcinlichwala** [comment]
Hello @rasikraj01!
I can see that you use different header names in access-control-allow-headers header:
`res.setHeader('Access-Control-Allow-Headers', 'x-access-token,content-type');`

...and in the login endpoint where you set the token header:
```
return user.generateAuthToken().then((token) => {
          res.header('x-auth', token).send(user);
        });
```

When there are different results in Postman and a browser it's usually a CORS issue.

Please let me know if it helps

**Comment by rasikraj01** [comment]
@marcinlichwala 
No changing <code>res.setHeader('Access-Control-Allow-Headers', 'x-access-token,content-type');</code>
to 
<code>res.setHeader('Access-Control-Allow-Headers', 'x-auth,content-type');</code>
did not help
<img width="1425" alt="screen shot 2017-10-18 at 10 20 36 am" src="https://user-images.githubusercontent.com/18498478/31701449-16f64b80-b3ee-11e7-803b-ddb3e0baa5b4.png">

**Comment by marcinlichwala** [comment]
Hello @rasikraj01,

I will clone the project later and take a look myself. 

There are 2 bits I would check before searching further:
- check which headers are visible in the network tab
- switch from axios to the native fetch() API and see if axios doesn't remove some headers from the response (don't have much experience with axios)

Cheers

**Comment by rasikraj01** [comment]
@marcinlichwala ,
Yeah, the x-auth header does show up in the network tab. So, what do I need to change in my code?

<img width="1440" alt="screen shot 2017-10-18 at 3 03 53 pm" src="https://user-images.githubusercontent.com/18498478/31711413-9bf774ea-b415-11e7-964c-e2b4cbe9439a.png">

**Comment by rasikraj01** [comment]
@marcinlichwala ?

**Comment by marcinlichwala** [comment]
You can check if installing a browser extension that disables CORS checks will help anything. That will give us some clues ;)
